src/pjpersist/datamanager.py

```diff
@@ -621,20 +625,32 @@ def flush(self, flush_hint=None):
         #
         # While writing objects, new sub-objects might be registered
         # that also need saving. We are flushing until nothing left to flush.
-        processed = set()
         while True:
             # Write every registered object, but make sure we write each
             # object just once.
-            todo = set(self._registered_objects.keys()) - processed
+            todo = set()
```

Comment:
nitpicking, move it one line below

Reply:
if I had to intuit, it's just moving it inside of `if flush_hint:` since without the hint we'll just create the set on its own.

Reply:
Ah, I'd rather define `todo` in one place and then populate it in both branches. Updated.